### PR TITLE
image preprocessing using dict input, and a test case.

### PR DESCRIPTION
**Pull request by jakesabathia2**
fix a bug that for an old tf-coreml,
passing image preprocessing parameters with dict input type won't work.
Also add a testcase.

### CODE REVIEW

File: tests/test_pretrained_models.py [hunk 1]
```diff
@@ -224,46 +224,108 @@ def _test_coreml_model_image_input(self, tf_model_path, coreml_model,
...
-    
+
     #Test the default CoreML evaluation
     coreml_out = coreml_model.predict(coreml_input, useCPUOnly = useCPUOnly)[coreml_output_name]
     coreml_out_flatten = coreml_out.flatten()
     self._compare_tf_coreml_outputs(tf_out_flatten, coreml_out_flatten)
 
-class TestModels(CorrectnessTest):         
-  
+class TestModels(CorrectnessTest):
```
Comment (jakesabathia2) [inline tests/test_pretrained_models.py]: Here matters.

File: tfcoreml/_tf_coreml_converter.py [hunk 2]
```diff
@@ -458,9 +458,23 @@ def _convert_pb_to_mlmodel(tf_model_path,
       if name in interface_blob_names:
         spec_layer.output[j] = name.replace(':', '__').replace('/', '__')
 
-  if image_input_names is not None:
-    for i, img in enumerate(image_input_names):
-      image_input_names[i] = img.replace(':', '__').replace('/', '__')
+  # replace ':' and '/' in input names by '__'
```
Comment (jakesabathia2) [inline tfcoreml/_tf_coreml_converter.py]: Here also matters.

**Comment by aseemw** [comment]
1. can you add a synthetic unit test a toy network in which there are two images as input, so that the dict is really put to work (say x1 and x2 are inputs, they are preprocessed and added that thats the output)
2. does this work when minimum_ios_deployement_target is '13'? Maybe not, as that would require a change in coremltools. In that case, please raise an error. (that error can be removed once coremltools adds support for it)

**Comment by jakesabathia2** [comment]
> * can you add a synthetic unit test (a toy network) in which there are two images as inputs, so that the dict is really put to work (say x1 and x2 are inputs, they are preprocessed and added and thats the output)
> * does this work when minimum_ios_deployement_target is '13'? Maybe not, as that would require a change in coremltools. In that case, please raise an error. (that error can be removed once coremltools adds support for it)

1. Will add it. Thanks!
2. Yes. `13` doesn't have this bug haha, since all `:` are removed.

**Comment by jakesabathia2** [comment]
@aseemw  @1duo  I just changed the testcase to test_tf_converter.

**Pull request merged**